Sources/AuthFoundation/OAuth2/OAuth2Client.swift

```diff
@@ -422,56 +422,65 @@ public final class OAuth2Client {
             }
         }
         
-        // Exchange the token
-        request.send(to: self) { result in
-            guard case let .success(response) = result else {
-                completion(result)
-                return
-            }
-            
+        var configuration = openIdConfiguration
+        if configuration == nil {
             group.enter()
             self.openIdConfiguration { result in
-                switch result {
-                case .success(let configuration):
-                    do {
-                        try response.result.validate(
-                            using: self,
-                            issuer: configuration.issuer,
-                            with: request as? IDTokenValidatorContext
-                        )
-                    } catch {
-                        completion(.failure(.validation(error: error)))
-                        return
-                    }
-                case .failure(let error):
-                    completion(.failure(.validation(error: error)))
+                defer { group.leave() }
+                if case let .success(response) = result {
+                    configuration = response
                 }
-                group.leave()
             }
+        }
         
-            // Wait for the JWKS keys, if necessary
-            group.notify(queue: DispatchQueue.global()) {
-                guard let idToken = response.result.idToken else {
+        group.notify(queue: DispatchQueue.global()) {
+            // Exchange the token
+            request.send(to: self) { result in
```

Comment:
There's no need to use `group.notify` here, because we want the token exchange request to happen in parallel with the jwks and openIdConfiguration requests. So let this one run immediately, and use the `group.notify` you have later in this method to handle the results.